pkg/config/types/v1alpha1/generate/generate.go

```diff
@@ -123,7 +124,15 @@ func (i *Input) GetControlPlaneEndpoint() string {
 // GetAPIServerSANs returns the formatted list of Subject Alt Name addresses for the API Server
 func (i *Input) GetAPIServerSANs() []string {
 	list := []string{"127.0.0.1", "::1"}
-	list = append(list, i.ControlPlaneEndpoint)
+
+	endpointURL, err := url.Parse(i.ControlPlaneEndpoint)
```

Comment:
Not necessarily for this PR, but I think it makes sense to make `ControlPlaneEndpoint` a `*net.URL`.

Reply:
totally agree. I'll drop a backlog ticket for this